docs/api/specs/webConnectionProtocol.md

```diff
@@ -0,0 +1,384 @@
+    T311 -->|other|T3113["reject with AgentError.InvalidFailover"]
+```
+
+## Providing Channel Selector and Intent Resolver UIs
+
+Users of FDC3 Desktop Agents often need access to UI controls that allow them to select user channels or to resolve intents that have multiple resolution options. Whilst apps can implement these UIs on their own via data and API calls provided by the `DesktopAgent` API, Desktop Agents typically provide these interfaces themselves.
+
+However, Browser Resident Desktop Agents may have difficulty displaying user interfaces over applications for a variety of reasons (inability to inject code, lack of permissions to display popups etc.), or may not (e.g. because they render applications in iframes within windows they control and can therefore display content over the iframe). The Web Connection Protocol and the `getAgent()` implementation based on it and incorporated into apps via the [`@finos/fdc3` npm module](https://www.npmjs.com/package/@finos/fdc3), is intended to help Desktop Agents deliver these UIs where necessary.
+
+```mermaid
+flowchart LR
+  subgraph DA [Desktop Agent Window]
+    A[(Desktop Agent)]
+  end
+  A-->B["getAgent()"]
+  subgraph App [App Window]
+    B
+    subgraph iframe1 [iframe 1]
+      cs[Channel Selector]
+    end
+    subgraph iframe2 [iframe 2]
+      ir[Intent Resolver]
+    end
+  end
+  B-->cs
+  B-->ir
+```
+
+The WCP allows applications to indicate to the `getAgent()` implementation whether they need the UIs (they may not need one or the other based on their usage of the FDC3 API, or because they implement UIs themselves) and for Desktop Agents to provide custom implementations of them, or defer to reference implementations provided by the FDC3 Standard. This is achieved via to following messages:
+
+- [`WCP1Hello`](https://fdc3.finos.org/schemas/next/api/WCP1Hello.schema.json): Sent by an application and incorporating boolean `payload.intentResolver` and `payload.channelSelector` fields, which are set to `false` if either UI is not needed (defaults to `true`).
+- [`WCP3Handshake`](https://fdc3.finos.org/schemas/next/api/WCP3Handshake.schema.json): Response sent by the Desktop Agent and incorporating `payload.intentResolverUrl` and `payload.channelSelectorUrl` fields, which should be set to the URL for each UI implementation that should be loaded into an iframe to provide the UI (defaults to URLs for reference UI implementations provided by the FDC3 project), or set to `false` to indicate that the respective UI is not needed. Setting these fields to `true` will cause the `getAgent()` implementation to use its default URLs representing a reference implementation of each UI.
+
+When UI iframes are created, the user interfaces may use the 'iframe' messages incorporated into the [Desktop Agent Communication Protocol (DACP)](./desktopAgentCommunicationProtocol#controlling-injected-user-interfaces) to communicate with the `getAgent()` implementation and through it the Desktop Agent.
```

Comment:
Following #1375 earlier "may use the 'iframe' messages" should probably now be "may use the `Fdc3UserInterface` messages"

Reply:
fixed in 976af80d5ba8d831e202aabcd37d1a7886e4c930